File: 파이썬/1427.py
'''
 * Author : JC
 * https://www.acmicpc.net/problem/1427
'''

# 버블 정렬
l = list(map(int,input()))
length = len(l)

for i in range(length-1, -1, -1) :
    for j in range(0, i) :
        if l[j] < l[j+1] :
            tmp = l[j]
            l[j] = l[j+1]
            l[j+1] = tmp

for i in l :
    print(i, end='')
             
# 선택 정렬
l = list(map(int,input()))
length = len(l)

for i in range(length-1) :
    key = i
    for j in range(i+1, length) :
         if l[key] < l[j] :  
             key = j 
    tmp = l[key]
    l[key] = l[i]
    l[i] = tmp
         
for i in l :
    print(i, end='')
    
    
# 삽입 정렬
l = list(map(int,input()))
length = len(l)

for i in range(1, length) :
    key = l[i] 
    j = i-1 
    while j>=0 and l[j] < key :
        l[j+1] = l[j]
        j -= 1 
    l[j+1] = key
    
for i in l :
    print(i, end='')

File: C++/02. 정렬 - 중급/1427.cpp
/*
 * Author : JC
 * https://www.acmicpc.net/problem/1427
*/

#include<iostream>

using namespace std;

int arr[1000000];
int n, idx;

int main(){

    cin >> n;

    while(n){
        arr[idx] = n%10;
        n /= 10;
        idx++;
    }
    
    for(int i=idx; i>=0; i--){
        for(int j=0; j<i; j++){
            if(arr[j] < arr[j+1]){
                int tmp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = tmp;
            }
        }
    }
    
    for(int i=0; i<idx; i++){
        cout << arr[i];
    }

    return 0;
}